packages/plugin-essentials/sources/commands/add.ts

```diff
@@ -202,6 +227,23 @@ export default class AddCommand extends BaseCommand {
           ]);
         }
       }
+
+      for (const key in suggestUtils.Target) {
+        const otherTarget = suggestUtils.Target[key as keyof typeof suggestUtils.Target];
```

Comment:
Shouldn't this be skipped if either `key` or `otherTarget` are `PEER`?

Reply:
Thanks for feedback!
Sorry for missing tests.
The main idea here was to clean up `PEER` (when using `--dev`, or when package exists in `devDependencies`), Because in all other scenarios either `dup === 'undefined'` or an error would be thrown (except for edge cases where a duplicate exists before calling `yarn add`).
OP expected yarn to auto clean duplicates. Do you think it's unnecessary?

Reply:
I think peers should be preserved when working with dev, because it's a valid use case: consider working on a plugin `babel-plugin-foo`. You'll want to list `babel` in your peer dependencies, but in order for your tests to be able to require it you'll also have to list it in your dev dependencies.

Reply:
Agree. So I removed the cleanup logic.